src/main/scheduler/scheduler.c

```diff
@@ -23,6 +23,7 @@
 
 #include "platform.h"
 
+#include "build/build_config.h"
```

Comment:
build_config.h has some "after target.h" clean-up defs:
```
// Cleanup dependency on I2C, USE_OLEDI2C and USE_DASHBOARD

#ifndef USE_I2C
# undef USE_OLEDI2C
#endif

#ifndef USE_OLEDI2C
# undef USE_DASHBOARD
#endif
```
The original code base always included bus_i2c_xxx.c regardless of presence/absence of USE_I2C. Likewise, display_hwgxxxx.c was included unconditionally by default. In this PR, I tried to sort this out.

I don't know if I can make the clean-up defs gone as USE_DASHBOARD is the default in target/common.h; let me think about it.